### PR TITLE
Added an input branch to initialise the project

### PR DESCRIPTION
**Pull request by SirriRyisa**
Added a dev and made it my default branch
Added an input branch to serve as my working branch

### CODE REVIEW

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello 👋🏽  @SirriRyisa

Good job working on the project so far! 
✔️ App was created using CRA
However, there are some issues that you need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] According to the requirements, please structure your app following the ducks pattern for your Redux file i.e.
``` ...
/src
    |---/components
    |---/redux
        |--- /books
                | books.js
        |--- /categories
                | categories.js
        | configureStore.js
    | index.js
```
- [ ] Your components should be working and should display the following according to the project requirements
Screenshot 2022-03-09 at 09 56 34

- [ ] Your README is currently just Microverse README template, please edit it to contain information about your project.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Books.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+
+const Books = () => (
+  <div className="main-container">
+    <div className="books-container" />
+
+  </div>
+);
+
+export default Books;
```
Comment (moise10r) [inline src/components/Books.js]: According to the project requirements:

image

Therefore:

- [ ] Kindly the `Books` component is the parent component of the `Book` component which mean This file will be responsible for rendering each book with the title and author taken as props.

Don't worry, I'll help you.

## Explanation

We need to create a reusable component that has each book's data.

## Hints

The said "Book.js" component should take the book data as a prop.  When done, include it inside your "Books" component as a child.

Look at this example:

### The new Book.js:

```JAVASCRIPT

// Eslint won't let you add props without these prop types validations.  
// In order to use them, you need to install them: npm install prop-types --save
import PropTypes from 'prop-types';  


// The new Book component.  Takes the data as props to render the title and author.
const Book = ({ title, author }) => ( 

      <div>
        <p>{title}</p>
        <button type="button">
          {author}
        </button>
      </div>
}

// After the function component, add validations to satisfy eslint.

BookItem.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
};

export default BookItem;
```

### On your Books.js:

```JAVASCRIPT

import Book from './Book';

const BooksList = () => {

  const booksList = [ // This is an example list of initial books.
    { id: 1, title: 'The Hobbit', author: 'J.R.R. Tolkien' },
    { id: 2, title: 'The Da Vinci Code', author: 'Dan Brown' },
    { id: 3, title: 'Pinocchio', author: 'Carlo Collodi' },
  ];

  return (
    <>

      <ul>
        {booksList.map((book) => ( // Iterate over the list of initial books.  
          <li key={book.id}>
            <Book title={book.title} author={book.author} /> // Here we are using the new imported Book component!!
          </li>
        ))}
      </ul>

    </>
  );
};

export default BooksList;
```

Now that you know how to do it, please include the "Book" component with props, in your "Books" component.  That way, we will meet this requirement 💪  


_Note: This is just an example to help you, and should not be taken as an "official way" to do it. This is just the result of my own personal effort and experience._

_Originally posted by @Meltrust in https://github.com/VTY1999/Bookstore/pull/1#discussion_r821565478_

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (moise10r) [inline src/App.js:1]: - [ ] Please follow the requirement given by the previous reviewer, I can see you didn't really understand, what was the requested change, what you need to do is: Create a folder inside your `src` folder called `redux` then create two more folders (`books and categories`) inside the redux folder after that you can now create a file inside these two folders, the first will be `books.js` in the` book` folder and the second ` categories. js` in `categories` folder
- see what the tree will look like:

``` ...
/src
    |---/components
    |---/redux
        |--- /books
                | books.js
        |--- /categories
                | categories.js
        | configureStore.js
    | index.js
```

**Review by moise10r — CHANGES_REQUESTED** [review]
### Hi @SirriRyisa

Great work on making the changes requested by a previous reviewer 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_





- [ ]  Your README looks great 👍🏻  but I think you can improve it by making it more professional. A professional README file should have a project title, project description, list of technology used in the project, instructions to set the project up locally,  and a screenshot of your project. please refer to this [template](https://github.com/microverseinc/readme-template), you can copy and paste the content but remember to customize it to your project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ OPTIONAL] Please, Write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. you can check the  [Professional repo - basic rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @moise10r when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by er-contreras — APPROVED** [review]
Hi @SirriRyisa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- I see that you did most of the work, well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**